Save the motion export selection in the local storage

reiterl opened the pull request:
Resolve #4999 

It needs some time to save the selection after closing the form and crMode is not saved.

reiterl commented on client/src/app/site/pages/meetings/pages/motions/pages/motion-export/components/motion-export/motion-export.component.ts:
```diff
@@ -257,20 +260,27 @@ export class MotionExportComponent extends BaseComponent implements AfterViewIni
     }
 
+    public override ngOnDestroy(): void {
+        this.storeService.set(`motion-export-selection`, this.dialogForm.value);
```
Done

reiterl commented on client/src/app/site/pages/meetings/pages/motions/pages/motion-export/components/motion-export/motion-export.component.ts:
```diff
@@ -257,20 +260,27 @@ export class MotionExportComponent extends BaseComponent implements AfterViewIni
+                this.dialogForm.patchValue(this.pdfDefaults);
+            }
         }
+        this.patchOnAnimation = true;
```
small rename and code change.

bastianjoel commented on client/src/app/site/pages/meetings/pages/motions/pages/motion-export/components/motion-export/motion-export.component.ts:
```diff
@@ -257,19 +260,28 @@ export class MotionExportComponent extends BaseComponent implements AfterViewIni
         this.isXLSXExport = this.fileFormats[tabChangeEvent.index] === ExportFileFormat.XLSX;
     };
 
     public animationDone(): void {
```
```suggestion
    public afterTabChanged(): void {
```

This was the confusing part. Please change before merging.

reiterl replied:
Renaming: Done.

bspekker reviewed (changes requested):
Two things do not work as they previously did:
* The text version should be saved. The previous selection should overwrite the standard in the settings.
* The selection for one file type should be saved even if a different file type was exported in the meantime.

reiterl commented:
The crMode (Text) is hardcoded selected at a special value, which was requested in the new motion export selection issue.
I won't change this. The old dialog did just save one selection of the form. I do the same here. If the saved selection between different file format worked before, it would be specific to the old form. I don't see an easy or fast way to implement  that in this new form.

reiterl commented:
Anyway, as the motion export form was reworked (See #4026 ) The saving of selection was **not** a requirement.
And the rework was merged without it.

bspekker commented:
The requirements of #4999 have changed since the implementation of #4026 
Please make the stated changes.

reiterl commented:
Updated the code.
Notice: you perhaps need to remove the old motion-export-selection object in the locale Storage, if you tested the old version.

The change includes

- save selections of all tabs 
- removed the special preselection of crMode/ltMode.
